### ISSUE TITLE
json_printer prints invalid number if number is very small

### ISSUE THREAD

**Issue by nyg** [pljson/pljson#69]
``` sql
DECLARE
  j json := json();
  c CLOB := ' ';
BEGIN
  j.put('a', 2.524867921197961e-49);
  j.path('a').to_clob(c, TRUE);
  dbms_output.put_line(c); -- 02.524867921197961e-49
END;
```

The issue is the 0 before the 2. This happens because with this specific number `TO_CHAR(number, 'TM9', 'NLS_NUMERIC_CHARACTERS=''.,''')` returns a number in scientific notation and as it is between 0 and 1 json_printer will add a 0 before the number (as it's expecting a number matching `\.\d+`).

The issue is at multiple places in json_printer (`ppea`, `ppmem`, `pretty_print_any`), so I've written a `to_json_number` function. I can do a pull request if you want...

``` sql
FUNCTION to_json_number(pi_number NUMBER) RETURN VARCHAR2 IS
  l_buffer VARCHAR2(4000);
BEGIN
  l_buffer := TO_CHAR(pi_number, 'TM9', 'NLS_NUMERIC_CHARACTERS=''.,''');

  IF -1 < pi_number AND pi_number < 0 AND SUBSTR(l_buffer, 1, 2) = '-.'
  THEN
    l_buffer := '-0' || SUBSTR(l_buffer, 2);
  ELSIF 0 < pi_number AND pi_number < 1 AND SUBSTR(l_buffer, 1, 1) = '.'
  THEN
    l_buffer := '0' || l_buffer;
  END IF;

  RETURN l_buffer;
END to_json_number;
```

In the if statements I still check if the number is between -1 and 0 (or 0 and 1) so if it's not it avoids doing calling `substr`.

**Comment by jsumners** [comment]
I wouldn't mind seeing a PR against the develop branch for this. Have you tried adjusting the regular expression to match `\d{1}?\.\d+`?

As for the `NLS_NUMERIC_CHARACTERS`, I don't believe you need the comma included. The RFC only allows for `.` -- https://tools.ietf.org/html/rfc7159#section-6

**Comment by nyg** [comment]
There is actually no regex in the code, mine was just as an example...

The comma is necessary otherwise I get a `numeric or value error`. The comma is actually already in the code, but apparently it doesn't get printed when using the TM9 format...

**Comment by jsumners** [comment]
👍 

I look forward to your fix.

**Comment by jsumners** [comment]
Solved by #72.
